protocols/primary-backup/src/main/java/io/atomix/protocols/backup/roles/BackupRole.java

```diff
@@ -167,6 +167,7 @@ private void requestRestore(NodeId primary) {
 
             context.service().restore(buffer);
             operations.clear();
+            context.setCommitIndex(response.index());
```

Comment:
why? isn't it redundant? we set the commit index a few lines above with context.resetIndex

Reply:
I do not know why `resetIndex` not set commitIndex, so i invoke setCommitIndex() here

Reply:
That's the only place that method is called, so it should be fine to remove this line and set it inside `resetIndex`